autoload/lsp.vim

```diff
@@ -148,28 +149,48 @@ function! s:unregister_events() abort
     doautocmd User lsp_unregister_server
 endfunction
 
+let s:event_queue = []
```

Comment:
Since vim doesn't have concept of private only to plugin, I added most of the private functions in same file.
Should we start moving to different file so it is easier to write tests as well maintain. We could prefix private functions with `_`

Reply:
Where should we move this? `lsp#utils` ?

Reply:
Seems like it can be a full different file `lsp#utils#_event_queue#add()`

Reply:
Hmm, ensure_flush is in script scope of lsp.vim . Seems not easier.

Reply:
Makes sense.